### ISSUE TITLE
Breakpoint ignored because generated code not found (source map problem?) not working yet

### ISSUE THREAD

**Issue by mrguzzi** [microsoft/vscode-chrome-debug#646]
Hello,
i struggled with this source mapping but i cannot resolve yet

that's my launch.json configuration:
```
{
"name": "Attach to Chrome",
"type": "chrome",
"request": "attach",
"port": 9222,
"webRoot": "${workspaceRoot}/rm-datadir-app",
"sourceMaps":true,
"sourceMapPathOverrides": {
                "webpack-internal:///": "${webRoot}/*",
              "webpack:///": "${webRoot}/*",
              "ng:///": "${webRoot}/*"
}
```

As I can see in the result of the .scripts command:

```
 webpack-internal:///./src/app/audit/audit.component.ts (c:\srdev\projects\rm0-ui\rm-datadir-app\src\app\audit\audit.component.ts)
    - webpack:\src\app\audit\audit.component.ts (c:\srdev\projects\rm0-ui\rm-datadir-app\src\app\audit\webpack:\src\app\audit\audit.component.ts)
```

once a breakpoint is hit, it stops into a read only file with this path:
`c:\srdev\projects\rm0-ui\rm-datadir-app\src\app\audit\webpack:\src\app\audit\audit.component.ts`

and not in the original source file. It is adding the \webpack:\src\app\audit\audit.component.ts path.

Here is my .scripts response:
https://gist.github.com/mrguzzi/417489863ae5d5205e52baa15c1af4e9

Here is log:
https://gist.github.com/mrguzzi/5f0e2213f6145600b4d8da931c07ac74

VS Code Version:
Version 1.19.1
Commit 0759f77bb8d86658bc935a10a64f6182c5a1eeba
Date 2017-12-19T09:46:23.884Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
Log file (set "trace": true in launch config):

What am I mistaking in remapping the source?

Thanks

**Comment by roblourens** [comment]
Can you share your project? Or how did you generate it?

```
SourceMap: sourceRoot: webpack:///
SourceMap: sources: ["./src/app/data-package/data-package-detail.component.ts"]
```

Your sourceMapPathOverrides don't look right. It should apply `'webpack:///./*': '${webRoot}/*'` by default, does it work if you remove all the sourceMapPathOverrides?

**Comment by mrguzzi** [comment]
@roblourens 
sorry but i cannot share my project
i generated with angular-cli
i tried substituting the sourcemap with yours or removing the source sourceMapPathOverrides but it works the same: it ignores the original source breakpoints and stops in a "read-only inlined content from source map" as it says on the tab of vsc and its path is the following:
C:\srdev\projects\rm0-ui\rm-datadir-app\src\app\audit\**webpack:**\src\app\audit\audit.component.ts

which is obviously the original source file
Thanks

**Comment by roblourens** [comment]
Can you remove the sourceMapPathOverrides, run again, and share the log from that?

**Comment by mrguzzi** [comment]
@roblourens 
Hello Rob,
i'm pretty sure that i run the .scripts without the overrides but i rerun it again updating the gist:
https://gist.github.com/mrguzzi/417489863ae5d5205e52baa15c1af4e9

the launch configuration was simply this:
 {
            "name": "Attach to Chrome",
            "type": "chrome",
            "request": "attach",
            "port": 9222,
            "webRoot": "${workspaceRoot}/rm-datadir-app"
        }

Thanks a lot

**Comment by roblourens** [comment]
I asked for the log because this seems like a case that should be handled automatically and I'm not sure why it isn't. Can you provide that (from `"trace": true`)? Thanks

**Comment by mrguzzi** [comment]
Hello @roblourens did you have any chance to check my trace log file?
Bye

**Comment by roblourens** [comment]
Yeah, I still don't know why it isn't working, so when I have time I need to create a similar project and try it out on windows to see if I can repro it.

**Comment by RyanMilligan** [comment]
I'm having a very similar problem where for some reason the debugger seems to be using the full path to the script as the name of the script when applying my `pathMapping` or `sourceMapPathOverrides` (the behavior is the same whichever I use).

My main project has a git submodule out to a shared library which is also installed as a local package using npm and referenced by the workspace, and I'm attempting to map the debugged scripts to the main enlistment of that shared library, like so:

```
"pathMapping": {
                "/node_modules/shared-library": "${workspaceRoot:shared-library}",
                "/": "${workspaceRoot}"
            }
```
When I run the `.scripts` command, all the entries for my shared library look like this:
```
http://localhost:62164/node_modules/shared-library/folder1/folder2/file.js!eval
    - http://localhost:62164/node_modules/shared-library/folder1/folder2/file.js (c:\project-root\my-app\shared-library\folder1\folder2\http:\localhost:62164\node_modules\shared-library\folder1/folder2/file.js)
```
I don't know if I'm seeing the exact same issue as the OP, but this is the only thread I've found that's mentioned that behavior of nesting the full URL inside the file path instead of just the file name. I've also used what seems to be the exact same scenario in another project and had it work fine, so I'm really not sure what to look at next.
